### PR TITLE
fix: messages to slack for archied toggles

### PR DESCRIPTION
**Pull request by ivarconr**
fixes: #740

### CODE REVIEW

File: snapshots/src/lib/addons/slack.test.js.md [hunk 1]
```diff
@@ -9,3 +9,9 @@ Generated by [AVA](https://avajs.dev).
 > Snapshot 1
 
     '{"username":"Unleash","icon_emoji":":unleash:","text":"some@user.com created feature toggle <http://some-url.com/#/features/strategies/some-toggle|some-toggle>\\n*Enabled*: no | *Type*: undefined | *Project*: undefined\\n*Activation strategies*: ```- name: default\\n```","channel":"#undefined","attachments":[{"actions":[{"name":"featureToggle","text":"Open in Unleash","type":"button","value":"featureToggle","style":"primary","url":"http://some-url.com/#/features/strategies/some-toggle"}]}]}'
+
+## Should call slack webhook for archived toggle
+
+> Snapshot 1
+
+    '{"username":"Unleash","icon_emoji":":unleash:","text":"The feature toggle *<http://some-url.com/#/archive/strategies/some-toggle|some-toggle>* was is *archived* by some@user.com.","channel":"#undefined","attachments":[{"actions":[{"name":"featureToggle","text":"Open in Unleash","type":"button","value":"featureToggle","style":"primary","url":"http://some-url.com/#/archive/strategies/some-toggle"}]}]}'
```
Comment (ivarconr) [inline snapshots/src/lib/addons/slack.test.js.md:17]: Looks like we have the correct message now, with link directly to the "archived" toggle.

File: src/lib/addons/slack.js [hunk 2]
```diff
@@ -80,23 +80,38 @@ class SlackAddon extends Addon {
...
 This was changed by ${createdBy}.`;
         }
-        return `The feature toggle *${feature}* was is *unmarked as stale*. This was changed by ${createdBy}.`;
+        return `The feature toggle *${feature}* was is *unmarked as stale* by ${createdBy}.`;
+    }
+
+    generateArchivedText(event) {
+        const { createdBy, data, type } = event;
+        const action = type === FEATURE_ARCHIVED ? 'archived' : 'revived';
+        const feature = `<${this.unleashUrl}/#/archive/strategies/${data.name}|${data.name}>`;
```
Comment (ivarconr) [inline src/lib/addons/slack.js:107]: I will fix this to use `featureLink" metod.

File: src/lib/addons/slack.js [hunk 3]
```diff
@@ -36,10 +36,10 @@ class SlackAddon extends Addon {
 
         let text;
 
-        if (event.type === FEATURE_STALE_ON) {
-            text = this.generateStaleText(event, true);
-        } else if (event.type === FEATURE_STALE_OFF) {
-            text = this.generateStaleText(event, false);
+        if ([FEATURE_ARCHIVED, FEATURE_REVIVED].includes(event.type)) {
+            text = this.generateArchivedText(event, true);
```
Comment (FredrikOseberg) [inline src/lib/addons/slack.js:40]: This method does not accept a second parameter.

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
Looks good. Perhaps just remove the unnecessary argument to the generateArchivedText.

File: src/lib/addons/slack.js [hunk 4]
```diff
@@ -80,23 +80,38 @@ class SlackAddon extends Addon {
...
         }
-        return `The feature toggle *${feature}* was is *unmarked as stale*. This was changed by ${createdBy}.`;
+        return `The feature toggle *${feature}* was is *unmarked as stale* by ${createdBy}.`;
+    }
+
+    generateArchivedText(event) {
+        const { createdBy, data, type } = event;
+        const action = type === FEATURE_ARCHIVED ? 'archived' : 'revived';
+        const feature = `<${this.featureLink(event)}|${data.name}>`;
+        return `The feature toggle *${feature}* was is *${action}* by ${createdBy}.`;
```
Comment (FredrikOseberg) [inline src/lib/addons/slack.js:108]:
```suggestion
        return `The feature toggle *${feature}* was *${action}* by ${createdBy}.`;
```

**Review by chriswk — APPROVED** [review]
LGTM

**Review by FredrikOseberg — APPROVED** [review]
Nice job.

**Pull request merged**